src/app/pages/MediaAssetPage/MediaAssetPage.jsx

```diff
@@ -52,10 +52,26 @@ const MediaAssetPage = ({ pageData }) => {
     ['relatedContent', 'groups', 0, 'promos'],
     pageData,
   );
-  const indexImagePath = path(['promo', 'indexImage', 'path'], pageData);
-  const indexImageLocator = indexImagePath
-    ? getImageParts(indexImagePath)[1]
-    : null;
+
+  const getIndexImageLocator = () => {
+    if (isLegacyMediaAssetPage(requestContext.canonicalLink)) {
+      return null;
+    }
+
+    const indexImagePath = path(['promo', 'indexImage', 'path'], pageData);
+
+    if (indexImagePath) {
+      const imageParts = getImageParts(indexImagePath);
+
+      if (imageParts && imageParts.length >= 2) {
+        return imageParts[1];
+      }
+    }
+
+    return null;
+  };
+
+  const indexImageLocator = getIndexImageLocator();
```

Comment:
Ramda could simplify this a lot, something like:

```javascript
const getIndexImageLocator = pipe(
  pathOr('', ['promo', 'indexImage', 'path']),
  getImageParts,
  last,
);

const indexImageLocator = isLegacyMediaAssetPage ? null : getIndexImageLocator(pageData);
```

Reply:
I'll give it a shot to see what happens 😬

Reply:
`pipe` didn't quite work - but I used `last` instead. I think it's _slightly_ easier to read now?